gtests/test_nopython_serializable.cpp

```diff
@@ -0,0 +1,131 @@
+#include <gtest/gtest.h>
+#include <thread>
+
+#ifdef Py_PYTHON_H
+#error "Python.h should not be included."
+#endif
+
+#include <modmesh/SerializableItem.hpp>
+namespace modmesh
+{
+
+namespace detail
+{
+
+struct Address : SerializableItem
+{
+    std::string country;
+    std::string city;
+    std::vector<std::string> phone_numbers;
+    std::vector<int> zip_codes;
+
+    MM_DECL_SERIALIZABLE(
+        register_member("country", country);
+        register_member("city", city);
+        register_member("phone_numbers", phone_numbers);
+        register_member("zip_codes", zip_codes);)
+};
+
+struct Pet : SerializableItem
+{
+    std::string name;
+    int age;
+    bool is_dog;
+    bool is_cat;
+
+    MM_DECL_SERIALIZABLE(
+        register_member("name", name);
+        register_member("age", age);
+        register_member("is_dog", is_dog);
+        register_member("is_cat", is_cat);)
+};
+
+struct Person : SerializableItem
```

Comment:
`Person`, `Address`, `Pet` objects are well designed for the different test cases, e.g. string array, object array, nested object.

Reply:
Is it currently limited to public member data?  If so, please add a comment somewhere in code.  If not, please build the test cases with private member data.

Reply:
added the test case for private members.